teuthology/dispatcher/__init__.py

```diff
@@ -0,0 +1,183 @@
+import logging
+import os
+import subprocess
+import sys
+import yaml
+
+from datetime import datetime
+
+from teuthology import setup_log_file, install_except_hook
+from teuthology import beanstalk
+from teuthology import report
+from teuthology.config import config as teuth_config
+from teuthology.repo_utils import fetch_qa_suite, fetch_teuthology
+from teuthology.lock.ops import block_and_lock_machines
+from teuthology.dispatcher import supervisor
+from teuthology.worker import prep_job
+from teuthology import safepath
+from teuthology.nuke import nuke
+
+log = logging.getLogger(__name__)
+start_time = datetime.utcnow()
+restart_file_path = '/tmp/teuthology-restart-dispatcher'
+stop_file_path = '/tmp/teuthology-stop-dispatcher'
+
+
+def sentinel(path):
+    if not os.path.exists(path):
+        return False
+    file_mtime = datetime.utcfromtimestamp(os.path.getmtime(path))
+    return file_mtime > start_time
+
+
+def restart():
+    log.info('Restarting...')
+    args = sys.argv[:]
+    args.insert(0, sys.executable)
+    os.execv(sys.executable, args)
+
+
+def stop():
+    log.info('Stopping...')
+    sys.exit(0)
+
+
+def load_config(archive_dir=None):
+    teuth_config.load()
+    if archive_dir is not None:
+        if not os.path.isdir(archive_dir):
+            sys.exit("{prog}: archive directory must exist: {path}".format(
+                prog=os.path.basename(sys.argv[0]),
+                path=archive_dir,
+            ))
+        else:
+            teuth_config.archive_base = archive_dir
+
+
+def main(args):
+    # run dispatcher in job supervisor mode if --supervisor passed
+    if args["--supervisor"]:
+        return supervisor.main(args)
+
+    verbose = args["--verbose"]
+    tube = args["--tube"]
+    log_dir = args["--log-dir"]
+    archive_dir = args["--archive-dir"]
```

Comment:
Can we add the default value for archive taken from teuthology config, that will make --archive-dir optional.